packages/eslint-plugin/src/rules/no-invalid-void-type.ts

```diff
@@ -174,6 +174,135 @@ export default createRule<[Options], MessageIds>({
       );
     }
 
+    function getMembers(
+      node: TSESTree.Node,
+    ):
+      | TSESTree.ClassElement[]
+      | TSESTree.ProgramStatement[]
+      | TSESTree.Statement[] {
+      switch (node.type) {
+        case AST_NODE_TYPES.ClassBody:
+        case AST_NODE_TYPES.Program:
+        case AST_NODE_TYPES.TSModuleBlock:
+        case AST_NODE_TYPES.BlockStatement:
+          return node.body;
+      }
+
+      /* istanbul ignore next */
+      return [];
+    }
+
+    function getParentFunctionDeclarationNode(
+      node: TSESTree.Node,
+    ): TSESTree.FunctionDeclaration | TSESTree.MethodDefinition | null {
+      let current = node.parent;
+      while (current) {
+        if (current.type === AST_NODE_TYPES.FunctionDeclaration) {
+          return current;
+        }
+
+        if (
+          current.type === AST_NODE_TYPES.FunctionExpression &&
+          current.parent.type === AST_NODE_TYPES.MethodDefinition
+        ) {
+          return current.parent;
+        }
+
+        current = current.parent;
+      }
+
+      return null;
+    }
+
+    function getFunctionDeclarationName(
+      node:
+        | TSESTree.FunctionDeclaration
+        | TSESTree.MethodDefinition
+        | TSESTree.TSDeclareFunction,
+    ): string | symbol | undefined {
+      if (
+        node.type === AST_NODE_TYPES.FunctionDeclaration ||
+        node.type === AST_NODE_TYPES.TSDeclareFunction
+      ) {
+        const id = nullThrows(
+          node.id,
+          'This may be null if and only if the parent is an `ExportDefaultDeclaration`.',
+        );
+
+        return id.name;
```

Comment:
This _currently_ only gets called for nodes whose parents aren't `ExportDefaultDeclaration,` and so this is never `null`. If this code gets refactored, it may change, and this can throw.

I can change this to an optional change-expression, though this will show up as a partially covered line by codecov.

```suggestion
        return node.id?.name;
```

Reply:
Similar to the other comment, my fallback would be to use a `!` with a disable comment explaining why the types are wrong. If possible, fixing up the types to be correct would be even better.

You might find the existing uses of the `MakeRequired` utils type interesting. Some places in code use for node types in similar situations. E.g. `no-confusing-void-expression` has handling for return statements that must have an `argument`. Maybe there's a similar flow here for functions that must have an `id`?

I'm not worried about refactors. Our test coverage is so high that the cost of unnecessary safety precautions in code is IMO higher than the cost of the brittleness.